build: require FFmpeg >= 6.1

mbouron opened the pull request:
The latest Ubuntu LTS release (24.04) ships with FFmpeg 6.1, so we can depend on it for the time being.

mbouron committed:
build: require FFmpeg >= 6.1

The latest Ubuntu LTS release (24.04) ships with FFmpeg 6.1, so we can depend
on it for the time being.

mbouron committed:
ci/msvc: bump ffmpeg version to 7.0

ubitux commented on src/mod_filtering.c:
```diff
@@ -157,12 +153,8 @@ static void audio_frame_to_sound_texture(struct filtering_ctx *ctx, AVFrame *dst
         bins[1] = bins[nb_samples];
-#else
-        av_rdft_calc(ctx->rdft, bins);
-#endif
```
The comment needs to be updated since it has duplicated description for the `av_rdft_calc` branch

mbouron replied:
Done.

ubitux commented on src/mod_decoding.c:
```diff
@@ -30,11 +30,7 @@
-    av_get_channel_layout_string(buf, buf_size, 0, par->channel_layout);
-#else
     av_channel_layout_describe(&par->ch_layout, buf, buf_size);
-#endif
```
This can now be inlined (can be done in a separate commit)

mbouron replied:
Done.

ubitux commented on src/mod_filtering.c:
```diff
@@ -157,12 +153,8 @@ static void audio_frame_to_sound_texture(struct filtering_ctx *ctx, AVFrame *dst
          */
-#if LIBAVUTIL_VERSION_INT >= AV_VERSION_INT(57, 19, 100)
         ctx->avtx_func(ctx->avtx, bins, bins, sizeof(float));
         bins[1] = bins[nb_samples];
```
We don't need to mimic `av_rdft_calc` behavior anymore, so this can be removed, the comment updated, and the magnitude computation below can be simplified (by removing the lowest frequency exception and starting the loop at `0`)

Also, with regards to `rdft_data`, I think it we can now:
- replace `FFTSample` with `float`
- ~~reduce the allocation size to `+1` instead of `+2`~~

mbouron replied:
Done in a separate commit.

ubitux commented on src/mod_filtering.c:
```diff
@@ -687,16 +664,9 @@ void nmdi_filtering_free(struct filtering_ctx **fp)
         av_freep(&ctx->window_func_lut);
         av_freep(&ctx->rdft_data[0]);
         av_freep(&ctx->rdft_data[1]);
-#if LIBAVUTIL_VERSION_INT >= AV_VERSION_INT(57, 19, 100)
         if (ctx->avtx) {
             av_tx_uninit(&ctx->avtx);
         }
-#else
```
Note: I'm under the impression that the condition is unnecessary

mbouron replied:
Removed in a separate commit.

ubitux commented on meson.build:
```diff
@@ -47,10 +47,10 @@ endif
+  dependency('libavformat', version: '>= 60.16.100'),
+  dependency('libavfilter', version: '>= 9.12.100'),
+  dependency('libavcodec', version: '>= 60.31.102'),
+  dependency('libavutil', version: '>= 58.29.100'),
```
This probably requires a changelog mention

mbouron replied:
Done.

mbouron committed:
mod_filtering: remove unnecessary check before calling av_tx_uninit()

mbouron committed:
mod_decoding: inline nmi_channel_layout_describe

mbouron committed:
mod_filtering: change rdft_data type to float

Since we do not rely on av_rdft anymore and that FFTSample is just a typedef
for float, we can switch the data type to float.

mbouron committed:
mod_filtering: stop mimicking av_rdft API

Since we no longer support the av_rdft API, we no longer need to mimic it.
